lib/sync/dataHandlers.js

```diff
@@ -259,6 +261,16 @@ module.exports = function(options) {
       handler(datasetId, dataHash, metaData, callback);
     },
 
+    /**
+     * @param datasetId
+     * @param metaData
+     * @param callback
+     */
+    countCollisions: function(datasetId, metaData, callback) {
```

Comment:
I think there are some problem here:

1. it adds a new function that the clients needs to override
2. it could return duplicated value - if some of the collisions are resolved.

I think a simpler approach is to just keep an counter on the dataset client, and then every time when there is a collision, we can increase that counter.